### PR TITLE
[WIP] Support netstandard1.5

### PR DESCRIPTION
**Pull request by galvesribeiro**
This is PR try to make Streamstone a `netstandard1.5` library so it can run in .Net Core.

The approach taken is basically the same we did in Orleans. 

For now, I created a new solution and replicated the 3 projects in it but using links to the files in the current directories just to facilitate the changes in it.

So, major changes:

1. In `netstandard` version of `WindowsAzure.Storage` there is no sync methods anymore, so whenever I saw a sync method, I replaced with the `XXXAsync()` related version. **suggestion**: I would remove all the Sync methods from the SS APIs and expose only the async ones and let the user call `.Wait()` or `.Result` in it.
2. The `Example` project is now a .Net Core console app -> We can multitarget it in case we really need but I guess it is not necessary
3. Some usages of `GetType().SomethingElse` where replaced to `GetType().GetTypeInfo().SomethingElse` since many methods were moved there.
4. I borrowed from #28 PR the `lambda -> string` queries since they are all good. Thanks @chrissimmons for already started that work.
5. The `Streastone` project is now a `netstandard1.5` project.
6. The test project is targeting `net462`. The reason for that, is that NUnit does not work with `dotnet` CLI yet and there is no estimated date to do so. I think the code on SS is very lean and simple, so at least for now, the tests running only on `net462` should be OK since there is no `forking` or `#if` whatsoever inside the code, so when NUnit got support to it, we could just migrate it.
7. NUnit was migrated to the latest version otherwise, it wouldn't be able to run tests on VS2017.

I need some help with NUnit just to make it detect the tests on VS TestExplorer and @aprooks will have a look in it later on. 

Besides that, the `Example` app run perfectly, so I appreciate @yevhen  and the others if I can have a review here.

image

Once you guys give an OK, I would revert/reset the changes at the PR and apply them to the main solution/projects, update the build scripts along with the .nuspec, and push the updates here.

Thanks! Looking forward to use SS in .Net Core with Orleans ASAP!

💃 

**PS**: Please note that appveyor build *will* fail. It will only pass once I update the main solution/project and the build scripts.

### CODE REVIEW

File: Source/Streamstone.Tests/Scenarios/Opening_stream.cs [hunk 1]
```diff
@@ -27,10 +27,10 @@ public void SetUp()
         }
         
         [Test]
-        [ExpectedException(typeof(StreamNotFoundException))]
-        public async void When_stream_does_not_exist()
+        //[ExpectedException(typeof(StreamNotFoundException))]
```
Comment (aprooks) [inline Source/Streamstone.Tests/Scenarios/Opening_stream.cs:30]: comment should be removed

File: Source/Streamstone/Stream.Operations.cs [hunk 2]
```diff
@@ -31,7 +31,7 @@ public Stream Execute()
 
                 try
                 {
-                    table.Execute(insert.Prepare());
+                    table.ExecuteAsync(insert.Prepare()).Wait();
```
Comment (khellang) [inline Source/Streamstone/Stream.Operations.cs:34]: These are deadlocks waiting to happen, right? Sync over async is as bad as it gets, especially in a library.

Comment (galvesribeiro) [reply Source/Streamstone/Stream.Operations.cs:34]: @khellang not exactly...

I'm actually in favor of remove all sync code and let the developer choose how to schedule it from the async methods...

Comment (galvesribeiro) [reply Source/Streamstone/Stream.Operations.cs:34]: I just tried to not change the current public APIs, but I still think the sync code should be removed...

Comment (khellang) [reply Source/Streamstone/Stream.Operations.cs:34]: http://blog.stephencleary.com/2012/07/dont-block-on-async-code.html

Comment (galvesribeiro) [reply Source/Streamstone/Stream.Operations.cs:34]: Yes, I know, that is why I'm saying that we should remove all the sync code from inside SS and let people decide where and how they should block or not the framework usage...

The only reason I did that, is the fact that I didn't want to change the public APIs yet but if you guys think we should, I'll be happy to do it.

Comment (khellang) [reply Source/Streamstone/Stream.Operations.cs:34]: Sure, my comment was simply to point out that as the code stands now, I don't think it's a good idea to pull it in. I'm all for removing sync APIs :+1:

Comment (galvesribeiro) [reply Source/Streamstone/Stream.Operations.cs:34]: @yevhen / @aprooks ?

**Comment by yevhen** [comment]
Ok, got some time to review the changes.

1) I don't have VS2017 on any of my machines and don't plan to install it until RTM. So first wrong thing - is that I can't open it in VS2015, which is bad :( I think other ppl could be in the same boat as me so we need to think how to handle this. If that could be only open in VS17, then it's better to create a separate branch, which will live until VS2017 RTM and all .NET Core releases will be delivered out of it. 
2) Since sync calls are not supported by Azure Core SDK then sync versions of methods need to be removed as well. It means that V1.x of Streamstone will stay on .NET 4.X and V2.x and above will be released exclusively for .NET Standard. In such way breaking changes and dependency trade-offs will be obvious to end-users and we won't sink in multi-targeting mess (until .NET xplat dust is set)
3) All projects including test project should have a single target (.NETStandard or whatever netcoreapp1, I compeltely lost in framework subset (re)brandings).
4) Build/release part need to be updated/intergrated. Build scripts (Nake.bat) need to be changed so all of the tasks work as they do with current version. Perhaps, they could simply call dotnet cli for some parts and do rest of the things (which are not currently supported by cli, such as running nunit tests)
5) I guess the problem with NUnit should go away once all projects are on the same target, plus running nunit test runner via Nake.bat task should solve the problem of dotnet cli support from NUnit. 

@galvesribeiro Does it make sense?

**Comment by yevhen** [comment]
The branch is https://github.com/yevhen/Streamstone/tree/v20netstd

**Comment by galvesribeiro** [comment]
@yevhen LGTM. I'll close this PR and make a new one to the new branch. I sill need help from someone that knows NUnit better than me.

Thanks

**Pull request closed without merging**